modules/user-interface-configuration/src/main/java/org/opencastproject/uiconfig/UIConfigRest.java

```diff
@@ -167,10 +167,11 @@ public Response getConfigFile(@PathParam("component") String component, @PathPar
 
     try {
       final String basePath = new File(uiConfigFolder, orgId).getCanonicalPath();
+      final String basePathWithSep = basePath.endsWith(File.separator) ? basePath : basePath + File.separator;
```

Comment:
Since `getCanonicalPath` should never end with a file separator, you can actually simplify your code:

```suggestion
      final String basePath = new File(uiConfigFolder, orgId).getCanonicalPath() + File.separator;
```